src/game/Tactical/Animation_Cache.cc

```diff
@@ -8,17 +8,22 @@
 
 #define EMPTY_CACHE_ENTRY 65000
 
-
-void AnimationSurfaceCacheType::init(ProfileID const pid)
+// Ensure cache state is valid even if init is never called
+// (the special soldiers created by Auto Resolve)
+AnimationSurfaceCacheType::AnimationSurfaceCacheType()
```

Comment:
This just screams for further simplification, as the whole ctor can be removed in favour of in-class initializers.

Reply:
Yeah, I don't like C++'s syntax for initializing an entire array to the same value too much, but it's probably still better than the separate constructor.

Reply:
For the second member you could've used default initialisation, but yeah, when you don't want zeros, it can get ugly without switching to std::array.